extensions/azurePublishNew/src/components/provisioningWizards/ImportResourcesWizard.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as React from 'react';
+import { usePublishApi } from '@bfc/extension-client';
+import formatMessage from 'format-message';
+
+import { WizardStep, Wizard } from '../shared/wizard';
+
+import { ImportInstructionsStep } from './steps';
+
+type Props = {
+  onStepChange: (stepId: string) => void;
+  stepIndex?: number;
+};
+export const ImportResourcesWizard = React.memo((props: Props) => {
+  const { stepIndex, onStepChange } = props;
+  const [steps, setSteps] = React.useState<WizardStep[]>([]);
+  const { setTitle, onBack } = usePublishApi();
+  const setDialogTitle = (step: WizardStep) => {
+    step && setTitle({ title: step.title, subText: step.subTitle });
+    return <></>;
+  };
+  React.useEffect(() => {
+    setSteps([
+      {
+        id: '1',
+        title: formatMessage('Configure resources to your publishing profile'),
+        subTitle: formatMessage('How would you like to provision Azure resources to your publishing profile?'),
+        onRenderContent: () => <ImportInstructionsStep />,
+        onBack,
+      },
+    ]);
+  }, []);
+  return (
+    <Wizard
+      firstStepId={stepIndex?.toString() ?? '1'}
```

Comment:
why stepIndex or 1, later on when stepIndex changes, firstStepId changes with it too, is that what you want?

Reply:
This was helpful earlier, when the wizard component was the child of  'ChooseProvisionAction', it was unmounting, It's not a issue anymore.